AirCasting/MainTabBarView.swift

```diff
@@ -54,14 +55,17 @@ struct MainTabBarView: View {
             
         }
         .onAppCameToForeground {
-            measurementUpdatingService.updateAllSessionsMeasurements()
+            measurementsDownloadingInProgress = true
+            measurementUpdatingService.updateAllSessionsMeasurements() { measurementsDownloadingInProgress = false }
         }
         .onAppear {
             UITabBar.appearance().backgroundColor = .aircastingBackground
             let appearance = UITabBarAppearance()
             appearance.backgroundImage = UIImage()
             appearance.shadowImage = UIImage.mainTabBarShadow
             UITabBar.appearance().standardAppearance = appearance
+            measurementsDownloadingInProgress = true
+            measurementUpdatingService.updateAllSessionsMeasurements() { measurementsDownloadingInProgress = false }
```

Comment:
doesn't Xcode warn about this state being set on a non-main thread? Because the completion here is called from a background queue

Reply:
Interestingly, it doesn't 🤔

Reply:
And what's more, when I add this dispatch, cards are not updated after measurements get downloaded. I have to toggle card expanding to see new time on the card.

Reply:
Or maybe not. Very misterious. :D